samples/jquery/00 ShoppingForm/src/modules/app/app.js

```diff
@@ -0,0 +1,152 @@
+import $ from 'jquery';
+import { productsService } from '../../services/productsService';
+import { productsFormValidation } from './validation/formProductValidationService';
+
+let storedProducts = [];
+let $selBrands, $selProducts, $formProducts, $txtNif, $txtDiscount;
+
+class App {
+  constructor() {
+    this.brands = [];
+
+    this.onBrandSelect = this.onBrandSelect.bind(this);
+    this.onFieldChange = this.onFieldChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+
+    $(this.init.bind(this));
+  }
+
+  init() {
+    this.loadBrands();
+    this.initializeSelectors();
+    this.setEventHandlers();
+  }
+
+  initializeSelectors() {
+    $selBrands = $('#selBrands');
+    $selProducts = $('#selProducts');
+    $txtNif = $('#txtNif');
+    $txtDiscount = $('#txtDiscount');
+    $formProducts = $('#formProducts');
+  }
+
+  loadBrands() {
+    productsService.fetchProducts()
+      .then(fetchedProducts => {
+        this.brands = fetchedProducts;
+        this.loadSelect($selBrands, this.brands);
+      });
+  }
+
+  loadSelect($select, items = []) {
+    $select.children().not(':first').remove();
+    const options = items
+      .map(item => `<option value="${item.id}">${item.description}</option>`)
+      .reduce((opts, option) => {
+        return `${opts}${option}`;
+      }, '');
+    $select.append(options);
+  }
+
+  setEventHandlers() {
+    $selBrands.change(this.onBrandSelect);
+    $txtNif.change(this.onFieldChange);
+    $txtDiscount.change(this.onFieldChange);
+    $txtDiscount.on('input', this.onDiscountType);
+    $formProducts.submit(this.onSubmit);
+  }
+
+  onBrandSelect(event) {
+    this.onFieldChange(event);
+    const product = Number(event.currentTarget.value) || null;
+    const isValidProduct = product !== null;
+    if (!isValidProduct) {
+      $selProducts.val('');
+    }
+    this.toggleAvailability($selProducts, isValidProduct);
+    this.loadProductsByBrand(product);
+  }
+
+  toggleAvailability($element, isEnabled) {
+    if (isEnabled) {
+      $element.removeAttr('disabled');
+    } else {
+      $element.attr('disabled', 'disabled');
+    }
+  }
+
+  loadProductsByBrand(id) {
+    if (id !== null) {
+      const osList = this.getProductsByBrand(id);
+      this.loadSelect($selProducts, osList);
+    }
+  }
+
+  getProductsByBrand(id) {
+    const brand = this.brands.filter(product => product.id === id);
+    return brand[0] && brand[0].products;
```

Comment:
Yes. If `brand[0]` does not exists because the filter didn't match it will return `undefined`. It's like:

```js
if(brand[0]) {
  return brand[0].products;
}
```

The point of this is to get `product[] | undefined` and pass it to `loadSelect($select, items = []` (it will initialize to `[]` if `items` is `undefined`.